src/seedu/addressbook/commands/Command.java

```diff
@@ -45,6 +45,8 @@ public CommandResult execute(){
 
     //Note: it is better to make the execute() method abstract, by replacing the above method with the line below:
     //public abstract CommandResult execute();
+    
+    public boolean isMutating(){ return false; };
```

Comment:
should there be default implementation for this method? Try recalling the Animal, Dog and Cat example I gave in class